sql/files/defaultdata/compare/compare.cc

```diff
@@ -176,16 +153,46 @@ int main(int argc, char **argv) {
 			if (!isfloat(team.c_str(), tval)) {
 				wrong_answer("Expected float, got: %s", team.c_str());
 			}
-			if (!equal(tval, jval, float_abs_tol, float_rel_tol)) {
-				wrong_answer("Too large difference.\n Judge: %s\n Team: %s\n Difference: %Lg\n (abs tol %Lg rel tol %Lg)",
-							 judge.c_str(), team.c_str(), fabsl(jval-tval), float_abs_tol, float_rel_tol);
+			/* Finite values are compared with some tolerance */
```

Comment:
Is there a way how this can still stay another method? Moving this into the main function is not the right place IMO.

I do like the improvement for the rest though.

Reply:
Fair point! I've rewritten the function to not return an `int` but to directly invoke `wrong_answer` with the correct message :slightly_smiling_face: